addons/mass_mailing/models/mail_mail.py

```diff
@@ -32,7 +32,9 @@ def _send_prepare_body(self):
 
                 parsed = werkzeug.urls.url_parse(url, scheme='http')
 
-                if parsed.scheme.startswith('http') and parsed.path.startswith('/r/'):
+                base_url_parsed = werkzeug.urls.url_parse(self.get_base_url(), scheme='http')
```

Comment:
Here, we probably want to normalize the base URLs before comparing them. (ex: `www.exemple.com` should probably be considered the same as `exemple.com`). There seem to be a helper funcion for that already:
https://github.com/odoo/odoo/blob/acbd2da7828090753a9ee3dc85c44bc8eca2aaa8/addons/website/tools.py#L190

A bit of a nitpick: the base URL parsing doesn't need to be inside the loop